pyanaconda/storage/partitioning.py

```diff
@@ -0,0 +1,146 @@
+        grow=False,
+        lv=True,
+        encrypted=True
+    )
+]
+
+# Partitioning requirements for workstations.
+WORKSTATION_PARTITIONING = [
+    PartSpec(
+        mountpoint="/",
+        size=Size("1GiB"),
+        max_size=Size("50GiB"),
+        grow=True,
+        btr=True,
+        lv=True,
+        thin=True,
+        encrypted=True),
+    PartSpec(
+        mountpoint="/home",
+        size=Size("500MiB"), grow=True,
+        required_space=Size("50GiB"),
+        btr=True,
+        lv=True,
+        thin=True,
+        encrypted=True),
+    PartSpec(
+        fstype="swap",
+        grow=False,
+        lv=True,
+        encrypted=True
+    )
+]
+
+
+def get_full_partitioning_requests(storage, platform, requests):
+    """Get the full partitioning requests.
+
+    :param storage: the Blivet's storage object
+    :param platform: the current platform object
+    :param requests: a list of partitioning specs
+    :return:
+    """
+    requests = _get_platform_specific_partitioning(platform, requests)
+    requests = _complete_partitioning_requests(storage, requests)
+    requests = _filter_default_partitions(requests)
+    return requests
+
+
+def _get_platform_specific_partitioning(platform, requests):
+    """Get the platform-specific partitioning.
+
+    The requests will be completed with the platform-specific
+    requirements.
+
+    :param platform: the current platform object
+    :param requests: a list of partitioning specs
+    """
+    return platform.set_default_partitioning() + requests
+
+
+def _complete_partitioning_requests(storage, requests):
+    """Complete the partitioning requests.
+
+    :param storage: the Blivet's storage object
+    :param requests: a list of partitioning specs
+    :return:
+    """
+    for request in requests:
+        if request.fstype is None:
+            request.fstype = storage.get_fstype(request.mountpoint)
+
+    return requests
+
+
+def _filter_default_partitions(requests):
+    """Filter default partitions based on the kickstart data.
+
+    :param requests: a list of requests
+    :return: a customized list of requests
+    """
+    auto_part_proxy = STORAGE.get_proxy(AUTO_PARTITIONING)
+    skipped_mountpoints = set()
+    skipped_fstypes = set()
+
+    # Create sets of mountpoints and fstypes to remove from autorequests.
+    if auto_part_proxy.Enabled:
+        # Remove /home if --nohome is selected.
+        if auto_part_proxy.NoHome:
+            skipped_mountpoints.add("/home")
+
+        # Remove /boot if --noboot is selected.
+        if auto_part_proxy.NoBoot:
+            skipped_mountpoints.add("/boot")
+
+        # Remove swap if --noswap is selected.
+        if auto_part_proxy.NoSwap:
+            skipped_fstypes.add("swap")
+
+            # Swap will not be recommended by the storage checker.
+            from pyanaconda.storage_utils import storage_checker
+            storage_checker.add_constraint(STORAGE_SWAP_IS_RECOMMENDED, False)
```

Comment:
I can help myself but I see this as a big side effect. Can you please change this to make this change more on purpose?

Reply:
This function will have to be changed anyway once it is moved to the storage module. I have added a `TODO` comment.

Reply:
Thanks, that is enough for me.